### PR TITLE
[analytics] initial draft

### PR DESCRIPTION
**Pull request by dcramer**
This is an initial abstraction for analytics within Sentry.

```python
>>> analytics.record('organization.created', organization)
```

- add sentry.analytics backend
- add sentry.utils.service.Service base helper
- add ``setup()`` call to initializers
- rename backend to service
- ensure ``Service.setup`` is called during tests

### CODE REVIEW

**Comment by dcramer** [comment]
this is still very earky -- just putting up here as a reminder and to let people leave feedback

**Comment by dcramer** [comment]
I want to try to get this in soon and then we'll work on an adapter to funnel into whatever event storage we want to use

**Comment by dcramer** [comment]
@mattrobenolt @tkaemming from an API perspective, any thoughts here? I want to get this in so we can start collecting better data.

**Comment by dcramer** [comment]
seems broken in getsnetry for some reason

**Comment by dcramer** [comment]
i made an additional set of changes to refactor some things, and improve how the services are setup (mostly to avoid complex recursive imports, by adding a ``setup()`` helper)

**Comment by dcramer** [comment]
seems to be passing in getsentry now with the recent changes

File: src/sentry/utils/services.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        Raise ``InvalidConfiguration`` if there is a configuration error.
+        """
+
+    def setup(self):
+        """
+        Initialize this service.
+        """
+
+
+class LazyServiceWrapper(LazyObject):
```
Comment (dcramer) [inline src/sentry/utils/services.py]: renamed from LazyBackendWrapper

File: CHANGES [hunk 2]
```diff
@@ -19,7 +21,10 @@ Schema Changes
 
 API Changes
 ~~~~~~~~~~~
+
 - Deprecate `dateStarted` in releases endpoints
+- Added data migration to create UserEmail objects for users whose primary emails did not have them
```
Comment (mattrobenolt) [inline CHANGES]: This is under the wrong version.

**Pull request merged**